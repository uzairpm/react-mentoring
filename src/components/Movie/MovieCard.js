import React, {Component} from 'react';
import { Link } from "react-router-dom";

import './moviecard.css';

export default class MovieCard extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        const { id, poster_path, title, release_date, genres } = this.props.movie;
        return (
            <div className="col-12 col-sm-6 col-md-4">
                <Link to={'/film/' + id} className="anchorStyle">
                    <div className="movieCard">
                        <img className="card-img-top albumArt" src={poster_path} />
                        <div className="movieDetails mt-2">
                            <div>
                                <label className="movieTitle">{title}</label>
                                <span className="badge badge-light float-right">{release_date.substr(0, 4)}</span>
                            </div>
                            <small>{genres.join(', ')}</small>
                        </div>
                    </div>
                </Link>
            </div>
        );
    }
}
