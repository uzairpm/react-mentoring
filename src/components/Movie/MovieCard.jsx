import React, {Component} from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

const StyledLink = styled(Link)`
    text-decoration: none;
    color: black;
    &:hover {
        text-decoration: none;
        color: black;
    }
`;
const MovieTitle = styled.label`
    color: #fa6c85;
    text-transform: uppercase;
    font-weight: 500;
    margin-bottom: 0;
`;
const MovieCardDiv = styled.div`
    padding: 40px;
    transition: box-shadow 0.3s ease-in-out;
    &:hover {
        box-shadow: 0px 0px 15px 2px rgba(250, 108, 133, 0.76);
    }
`;

export default class MovieCard extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        const {
            id,
            poster_path,
            title,
            release_date,
            genres
        } = this.props.movie;
        return (
            <div className="col-12 col-sm-6 col-md-4">
                <StyledLink to={'/film/' + id} className="anchorStyle">
                    <MovieCardDiv>
                        <img className="card-img-top rounded" src={poster_path} alt="Movie artwork" />
                        <div className="movieDetails mt-2">
                            <div>
                                <MovieTitle>{title}</MovieTitle>
                                <span className="badge badge-light float-right">{release_date.substr(0, 4)}</span>
                            </div>
                            <small>{genres.join(', ')}</small>
                        </div>
                    </MovieCardDiv>
                </StyledLink>
            </div>
        );
    }
}
