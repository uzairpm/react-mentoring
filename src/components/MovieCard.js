import React, {Component} from 'react';

export default class MovieCard extends Component {
    constructor(props) {
        super(props);
        this.state = {

        };
    }
    render() {
        const { poster_path, title, release_date, genres } = this.props.movie;
        return (
            <div className="col-12 col-sm-6 col-md-4">
                <div className="movieCard">
                    <img className="card-img-top albumArt"
                        src={poster_path} />
                    <div className="movieDetails">
                        <div>
                            <label className="movieTitle">{title}</label>
                            <span className="badge badge-light float-right">{release_date.substr(0, 4)}</span>
                        </div>
                        <small>{genres.join(', ')}</small>
                    </div>
                </div>
            </div>
        );
    }
}