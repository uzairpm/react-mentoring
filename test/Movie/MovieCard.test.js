import React from 'react';

import MovieCard from '../../src/components/Movie/MovieCard';

describe('Check Component creation', () => {
    test('MovieCard component', () => {
        const movieObj = {
            id: 101,
            poster_path: '',
            title: 'Feel Good',
            vote_average: 8,
            genres: ['Action', 'Drama'],
            release_date: '2017-10-10',
            runtime: 108,
            overview: 'A feel good movie'
        };
        const element = shallow(<MovieCard movie={movieObj}/>);
        // console.log(element.debug());
        expect(element).toMatchSnapshot();
        expect(element.find('span').text()).toEqual('2017');
    });
});