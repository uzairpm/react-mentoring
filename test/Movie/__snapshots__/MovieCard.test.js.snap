// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation MovieCard component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    movie={
      Object {
        "genres": Array [
          "Action",
          "Drama",
        ],
        "id": 101,
        "overview": "A feel good movie",
        "poster_path": "",
        "release_date": "2017-10-10",
        "runtime": 108,
        "title": "Feel Good",
        "vote_average": 8,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="anchorStyle"
        replace={false}
        to="/film/101"
      >
        <div
          className="movieCard"
        >
          <img
            className="card-img-top albumArt"
            src=""
          />
          <div
            className="movieDetails mt-2"
          >
            <div>
              <label
                className="movieTitle"
              >
                Feel Good
              </label>
              <span
                className="badge badge-light float-right"
              >
                2017
              </span>
            </div>
            <small>
              Action, Drama
            </small>
          </div>
        </div>
      </Link>,
      "className": "col-12 col-sm-6 col-md-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="movieCard"
        >
          <img
            className="card-img-top albumArt"
            src=""
          />
          <div
            className="movieDetails mt-2"
          >
            <div>
              <label
                className="movieTitle"
              >
                Feel Good
              </label>
              <span
                className="badge badge-light float-right"
              >
                2017
              </span>
            </div>
            <small>
              Action, Drama
            </small>
          </div>
        </div>,
        "className": "anchorStyle",
        "replace": false,
        "to": "/film/101",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="card-img-top albumArt"
              src=""
            />,
            <div
              className="movieDetails mt-2"
            >
              <div>
                <label
                  className="movieTitle"
                >
                  Feel Good
                </label>
                <span
                  className="badge badge-light float-right"
                >
                  2017
                </span>
              </div>
              <small>
                Action, Drama
              </small>
            </div>,
          ],
          "className": "movieCard",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "card-img-top albumArt",
              "src": "",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <label
                    className="movieTitle"
                  >
                    Feel Good
                  </label>
                  <span
                    className="badge badge-light float-right"
                  >
                    2017
                  </span>
                </div>,
                <small>
                  Action, Drama
                </small>,
              ],
              "className": "movieDetails mt-2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      className="movieTitle"
                    >
                      Feel Good
                    </label>,
                    <span
                      className="badge badge-light float-right"
                    >
                      2017
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Feel Good",
                      "className": "movieTitle",
                    },
                    "ref": null,
                    "rendered": "Feel Good",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2017",
                      "className": "badge badge-light float-right",
                    },
                    "ref": null,
                    "rendered": "2017",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Action, Drama",
                },
                "ref": null,
                "rendered": "Action, Drama",
                "type": "small",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="anchorStyle"
          replace={false}
          to="/film/101"
        >
          <div
            className="movieCard"
          >
            <img
              className="card-img-top albumArt"
              src=""
            />
            <div
              className="movieDetails mt-2"
            >
              <div>
                <label
                  className="movieTitle"
                >
                  Feel Good
                </label>
                <span
                  className="badge badge-light float-right"
                >
                  2017
                </span>
              </div>
              <small>
                Action, Drama
              </small>
            </div>
          </div>
        </Link>,
        "className": "col-12 col-sm-6 col-md-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="movieCard"
          >
            <img
              className="card-img-top albumArt"
              src=""
            />
            <div
              className="movieDetails mt-2"
            >
              <div>
                <label
                  className="movieTitle"
                >
                  Feel Good
                </label>
                <span
                  className="badge badge-light float-right"
                >
                  2017
                </span>
              </div>
              <small>
                Action, Drama
              </small>
            </div>
          </div>,
          "className": "anchorStyle",
          "replace": false,
          "to": "/film/101",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="card-img-top albumArt"
                src=""
              />,
              <div
                className="movieDetails mt-2"
              >
                <div>
                  <label
                    className="movieTitle"
                  >
                    Feel Good
                  </label>
                  <span
                    className="badge badge-light float-right"
                  >
                    2017
                  </span>
                </div>
                <small>
                  Action, Drama
                </small>
              </div>,
            ],
            "className": "movieCard",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "card-img-top albumArt",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <label
                      className="movieTitle"
                    >
                      Feel Good
                    </label>
                    <span
                      className="badge badge-light float-right"
                    >
                      2017
                    </span>
                  </div>,
                  <small>
                    Action, Drama
                  </small>,
                ],
                "className": "movieDetails mt-2",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label
                        className="movieTitle"
                      >
                        Feel Good
                      </label>,
                      <span
                        className="badge badge-light float-right"
                      >
                        2017
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Feel Good",
                        "className": "movieTitle",
                      },
                      "ref": null,
                      "rendered": "Feel Good",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2017",
                        "className": "badge badge-light float-right",
                      },
                      "ref": null,
                      "rendered": "2017",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Action, Drama",
                  },
                  "ref": null,
                  "rendered": "Action, Drama",
                  "type": "small",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
